### ISSUE TITLE
removeFile don't remove line into database

### ISSUE THREAD

**Issue by afiocre** [Baseflow/flutter_cache_manager#268]
## 🐛 Bug Report

When i call the `removeFile()` i see that the cached file is deleted into `/data/user/0/fr.serialtrip/cache/libCachedImageData`
But not into the database **libCachedImageData.db**

When I see the code, i don't understand. Is it a bug ?

> Future<void> removeCachedFile(CacheObject cacheObject) async {
    final provider = await _cacheInfoRepository;
    final toRemove = <int>[];
    unawaited(_removeCachedFile(cacheObject, toRemove));
    await provider.deleteAll(toRemove);
  }

With this code `unawaited(_removeCachedFile(cacheObject, toRemove));` it's impossible to delete the IDs from database (SQLite in my case) because it's not an await. So the toRemove is empty on the next line.

### Configuration

**Version:** 2.1.1

**Platform:** 
- [x] :robot: Android

**Comment by renefloor** [comment]
That part starts with adding the ID to the list, so that doesn't need to be awaited.
```
  Future<void> _removeCachedFile(
      CacheObject cacheObject, List<int> toRemove) async {
    if (toRemove.contains(cacheObject.id)) return;

    toRemove.add(cacheObject.id);
```

I don't know what is going wrong here.

**Comment by demos77** [comment]
No, the last instruction seems to delete from SQLite database the line whith ID from ToRemove.

But in the _removeCachedFile it's to remove the physical file and with an unwait never the toRemove is fill for provider.deleteAll()

**Comment by demos77** [comment]
2 tests:

First **FAIL**
```
Future<void> _removeCachePhotoTrip(int tripId) async {
	final Trip oldTrip = await getTripFromSQLite(tripId);
	if (null != oldTrip) {
	  await _removeCachePhoto(oldTrip.photo);
	  for (Day day in oldTrip.days) {
		for (Step step in day.steps) {
		  await _removeCachePhoto(step.photo);
		}
	  }
	}
}
```
The image key is NOT DELETE from SQLite.

Second **SUCESS**
```
Future<void> _removeCachePhotoTrip(int tripId) async {
	final Trip oldTrip = await getTripFromSQLite(tripId);
	if (null != oldTrip) {
	  await _removeCachePhoto(oldTrip.photo);
	  for (Day day in oldTrip.days) {
		for (Step step in day.steps) {
		  await _removeCachePhoto(step.photo);
		  var test = 'https://serialtrip.fr/media/cache/photo_web/media/places/55/photo-408.jpg';
		  await _removeCachePhoto(test);
		  print(step.photo);
		  print(test);
		  print(step.photo == test);
		}
	  }
	}
}
```
```
I/flutter (17725): https://serialtrip.fr/media/cache/photo_web/media/places/55/photo-408.jpg
I/flutter (17725): https://serialtrip.fr/media/cache/photo_web/media/places/55/photo-408.jpg
I/flutter (17725): true
```
The image key is IS DELETE from SQLite. but with the instruction with test var (direct String).
From the object step.photo not ... But we see on the debug console that's the same String with TRUE in the compare.

Help because for me it's very very space / strange :(

**Comment by demos77** [comment]
Ok my last test give the way to the bug ...

My function to remove the file from cache is:

```
  Future<void> _removeCachePhoto(String path) async {
    await DefaultCacheManager().removeFile(path);
  }
```

And when we call this into a loop FOR it's delete the file but not the ID into de SQLite database.

If i duplicate the removeFile(), it's fix the bug ! But why we need this ?
You can reproduce easy the bug into a FOR loop.

```
  Future<void> _removeCachePhoto(String path) async {
    await DefaultCacheManager().removeFile(path);
    await DefaultCacheManager().removeFile(path);
  }
```

**Comment by renefloor** [comment]
What do you mean with a for loop? Removing like 10 different files?

**Comment by demos77** [comment]
Yep, i have a Trip who containt X days who contain Y step.
And i have a function to remove all photos from cache for cleaning.
So i have a loop for steps who each have a different photo.

So the loop try to remove all the photo of  steps and there is a bug.
If i do each after each it's OK.
but the same in the loop it's NOT OK.

easy to reprocude.

**Comment by renefloor** [comment]
Do you do `await _removeCachePhoto(String path)` in the loop, or just `_removeCachePhoto(String path)`?

**Comment by demos77** [comment]
I have try both, but same result :(

**Comment by marcuslindemannrohden** [comment]
> I have try both, but same result :(

are you using a foreach loop? if yes, it ignores await's. 
You would need to iterate over a normal indexed for loop to be able to use await properly.

**Comment by bohdan1krokhmaliuk** [comment]
@renefloor @demos77  
Same issue for IOS.

Looks like CacheObject sometimes may have no ID property - that's why `toRemove` map is `int:null` map and that's why the record is not deleted from the database.
![image](https://user-images.githubusercontent.com/59435984/110522670-cee1c480-8119-11eb-8956-aac59bdc2bb3.png)
![image](https://user-images.githubusercontent.com/59435984/110522704-db661d00-8119-11eb-8cab-fe143b753583.png)

**Comment by bohdan1krokhmaliuk** [comment]
UPD: by the way if you try to delete object with the same key for the second time - most likely that `CacheObject` will have an Id and the database will be updated. 
![image](https://user-images.githubusercontent.com/59435984/110525193-fb4b1000-811c-11eb-8eba-e7d9154ae0b7.png)
